diff --git a/adb.cpp b/adb.cpp
index f397da0..364ad45 100644
--- a/adb.cpp
+++ b/adb.cpp
@@ -96,10 +96,10 @@ std::string adb_version() {
     // Don't change the format of this --- it's parsed by ddmlib.
     return android::base::StringPrintf(
         "Android Debug Bridge version %d.%d.%d\n"
-        "Version %s-%s\n"
+        "Version %s\n"
         "Installed as %s\n",
         ADB_VERSION_MAJOR, ADB_VERSION_MINOR, ADB_SERVER_VERSION,
-        PLATFORM_TOOLS_VERSION, android::build::GetBuildNumber().c_str(),
+        PLATFORM_TOOLS_VERSION,
         android::base::GetExecutablePath().c_str());
 }
 
diff --git a/adb_mdns.cpp b/adb_mdns.cpp
index 087a9ab..5038156 100644
--- a/adb_mdns.cpp
+++ b/adb_mdns.cpp
@@ -23,6 +23,9 @@
 #include <android-base/stringprintf.h>
 #include <android-base/strings.h>
 
+#include <atomic>
+#include <cstring>
+
 #include "adb_trace.h"
 
 #define ADB_SECURE_SERVICE_VERSION_TXT_RECORD(ver) ("v=" #ver)
diff --git a/adb_mdns.h b/adb_mdns.h
index 471ec25..1868899 100644
--- a/adb_mdns.h
+++ b/adb_mdns.h
@@ -19,6 +19,7 @@
 #include <optional>
 #include <string>
 #include <string_view>
+#include <functional>
 
 // The rules for Service Names [RFC6335] state that they may be no more
 // than fifteen characters long (not counting the mandatory underscore),
