diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index aadcb9b..b95462c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -567,10 +567,10 @@ endif()
 
 # Add minimal googletest targets. The provided one has many side-effects, and
 # googletest has a very straightforward build.
-add_library(boringssl_gtest third_party/googletest/src/gtest-all.cc)
-target_include_directories(boringssl_gtest PRIVATE third_party/googletest)
+# add_library(boringssl_gtest third_party/googletest/src/gtest-all.cc)
+# target_include_directories(boringssl_gtest PRIVATE third_party/googletest)
 
-include_directories(third_party/googletest/include)
+# include_directories(third_party/googletest/include)
 
 # Declare a dummy target to build all unit tests. Test targets should inject
 # themselves as dependencies next to the target definition.
