diff --git a/fastboot/super_flash_helper.cpp b/fastboot/super_flash_helper.cpp
index b617ce8..ffde61b 100644
--- a/fastboot/super_flash_helper.cpp
+++ b/fastboot/super_flash_helper.cpp
@@ -19,6 +19,7 @@
 #include <android-base/logging.h>
 
 #include "util.h"
+#include <climits>
 
 using android::base::borrowed_fd;
 using android::base::unique_fd;
diff --git a/libpackagelistparser/packagelistparser.cpp b/libpackagelistparser/packagelistparser.cpp
index 59c3a74..083efb8 100644
--- a/libpackagelistparser/packagelistparser.cpp
+++ b/libpackagelistparser/packagelistparser.cpp
@@ -22,12 +22,16 @@
 #include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/limits.h>
+// #include <sys/limits.h>
+#include <limits.h>
 
 #include <memory>
 
 #include <log/log.h>
 
+# define GID_MAX UINT_MAX
+# define UID_MAX UINT_MAX
+
 static bool parse_gids(const char* path, size_t line_number, const char* gids, pkg_info* info) {
   // Nothing to do?
   if (!gids || !strcmp(gids, "none")) return true;
diff --git a/libutils/Errors.cpp b/libutils/Errors.cpp
index 74f3bef..5198121 100644
--- a/libutils/Errors.cpp
+++ b/libutils/Errors.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include <utils/Errors.h>
+#include <string.h>
 
 namespace android {
 
