diff --git a/CMakeLists.txt b/CMakeLists.txt
index 04cb330..ab00478 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,7 @@ endif()
 if (CYGWIN)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
 else()
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 17)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
@@ -194,7 +194,7 @@ set(protobuf_LINK_LIBATOMIC false)
 if (NOT MSVC)
   include(CheckCXXSourceCompiles)
   set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
-  set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -std=c++11)
+  set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -std=c++17)
   check_cxx_source_compiles("
     #include <atomic>
     int main() {
@@ -298,7 +298,8 @@ endif (MSVC)
 include_directories(
   ${ZLIB_INCLUDE_DIRECTORIES}
   ${protobuf_BINARY_DIR}
-  ${protobuf_SOURCE_DIR}/src)
+  ${protobuf_SOURCE_DIR}/src
+  ${protobuf_SOURCE_DIR}/config)
 
 if (protobuf_UNICODE)
   add_definitions(-DUNICODE -D_UNICODE)
diff --git a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
index 83e9703..d3a5ad0 100644
--- a/cmake/libprotobuf-lite.cmake
+++ b/cmake/libprotobuf-lite.cmake
@@ -97,12 +97,18 @@ if(protobuf_HAVE_LD_VERSION_SCRIPT)
   set_target_properties(libprotobuf-lite PROPERTIES
     LINK_DEPENDS ${protobuf_SOURCE_DIR}/src/libprotobuf-lite.map)
 endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+target_compile_options(libprotobuf-lite PRIVATE
+    -std=c++17
+    -stdlib=libc++
+    )
+endif()
 target_link_libraries(libprotobuf-lite PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf-lite PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf-lite PRIVATE log)
+  target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
diff --git a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
index 07e4bcf..354991b 100644
--- a/cmake/libprotobuf.cmake
+++ b/cmake/libprotobuf.cmake
@@ -110,6 +110,12 @@ if(protobuf_HAVE_LD_VERSION_SCRIPT)
   set_target_properties(libprotobuf PROPERTIES
     LINK_DEPENDS ${protobuf_SOURCE_DIR}/src/libprotobuf.map)
 endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+target_compile_options(libprotobuf PRIVATE
+    -std=c++17
+    -stdlib=libc++
+    )
+endif()
 target_link_libraries(libprotobuf PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 if(protobuf_WITH_ZLIB)
   target_link_libraries(libprotobuf PRIVATE ${ZLIB_LIBRARIES})
@@ -118,7 +124,7 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf PRIVATE log)
+  target_include_directories(libprotobuf PRIVATE ${protobuf_SOURCE_DIR}/android)
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
diff --git a/cmake/libprotoc.cmake b/cmake/libprotoc.cmake
index 15a47e5..392640c 100644
--- a/cmake/libprotoc.cmake
+++ b/cmake/libprotoc.cmake
@@ -121,6 +121,12 @@ if(protobuf_HAVE_LD_VERSION_SCRIPT)
   set_target_properties(libprotoc PROPERTIES
     LINK_DEPENDS ${protobuf_SOURCE_DIR}/src/libprotoc.map)
 endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+target_compile_options(libprotoc PRIVATE
+    -std=c++17
+    -stdlib=libc++
+    )
+endif()
 target_link_libraries(libprotoc PRIVATE libprotobuf)
 if(protobuf_BUILD_SHARED_LIBS)
   target_compile_definitions(libprotoc
